### PR TITLE
[Doc] pytest documentation

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

Documentation on unit testing in pyDARN using pytest.

## Approval

**Number of approvals:** 1-2 proof reading

## ReadTheDocs link:

https://pydarn.readthedocs.io/en/ehn-update_docs_with_ptest/

### CODE REVIEW

File: docs/dev/PR.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Before submitting a PR check the code check list first:
 
 - copyright and disclaimer, or modification line is added 
 - the code installs with python 3.6 and higher 
+- the code doesn't fail on [`pytest` test](pytest.md)
+- Any new features need to have [tests for `pytest` to run on](pytest.md)
```
Comment (carleyjmartin) [inline docs/dev/PR.md:28]:
```suggestion
- any new features need to have [tests for `pytest` to run on](pytest.md)
```

File: docs/dev/branching.md [hunk 2]
```diff
@@ -66,4 +66,4 @@ In git, "branches" are used to make changes to the code without affecting the ma
         git push origin <branch name>
 
 * Repeat the above commands above as you work on the code changes 
-* Once you have completed, documented, and tested your code then you can create a pull request, see [pull request](PR.md)
+* Once you have completed, documented, tested and update the [unit tests](pytest.md) then you can create a pull request, see [pull request](PR.md)
```
Comment (carleyjmartin) [inline docs/dev/branching.md:69]:
```suggestion
* Once you have completed, documented, tested and updated the [unit tests](pytest.md) then you can create a pull request, see [pull request](PR.md)
```

File: docs/dev/pytest.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+This version of the GNU Lesser General Public License incorporates the terms
+and conditions of version 3 of the GNU General Public License, supplemented by
+the additional permissions listed below.
+-->
+
+## Unit Testing
+
+Unit testing in pyDARN is a way to check that pydarn is running correctly with any new changes to the code. These tests are meant to be simple and short to only test input (parameter) changes to ensure the code does not fail. 
+Exceptions can also be tested to make sure it fails correctly. Please note testing in pyDARN is for validity (code runs as it should) but not validation (outputting accurate plots). Validation testing is done by users who understanding the scientific accuracy the plots represent.
```
Comment (carleyjmartin) [inline docs/dev/pytest.md:18]:
```suggestion
Exceptions can also be tested to make sure it fails correctly. Please note testing in pyDARN is for validity (code runs as it should) but not validation (outputting accurate plots). Validation testing is done by users who understand the scientific accuracy the plots represent. 
```

File: docs/dev/pytest.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Running `pytest` in pyDARN is for developers and testers that work with GitHub. To run `pytest` in pyDARN:
+
+!!! Warning
+    Make sure you install pytest in your virtual environment or on your computer
+
+1. [clone the pydarn repository](https://github.com/superdarn/pydarn)
+2. change directory to pyDARN `cd pydarn`
+3. install pydarn `pip install . --user` 
+    - you can install a [virtual environment](../users/install.md) first to protect your current pyDARN install 
+4. now run pytest `pytest` it should report no fails
```
Comment (carleyjmartin) [inline docs/dev/pytest.md:33]:
```suggestion
1. [Clone the pydarn repository](https://github.com/superdarn/pydarn)
2. Change directory to pyDARN `cd pydarn`
3. Install pydarn `pip install . --user` 
    - you can install a [virtual environment](../users/install.md) first to protect your current pyDARN install 
4. Now run pytest `pytest` it should report no fails 
```

File: docs/dev/pytest.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+You can use `test/test_rtp.py` as a template on how to write test code and read [`pytest`](https://docs.pytest.org/en/6.2.x/) on other plugins to for testing if its needed. 
+
+Make sure to add what is needed to test and not too many parameterization options as it will make the tests go slow. Keep it to bare minimum needed to the test the parameter input. 
+Remember this is not to test the accuracy as pytest cannot look at the plot but rather to test edge cases of parameters that make it fail.
+
+!!! Note
+    There are files provided for testing in the `test/data/` directory. If you need to add please keep the file size as small as possible and check in with other before adding. If its too big, it may not be able to be added to GitHub and cause pydarn to be slow to download. 
+
+To avoid writing several tests with the same body of code, consider using the `paranetrize` to provide various types of inputs for one function.
```
Comment (carleyjmartin) [inline docs/dev/pytest.md:59]:
```suggestion
To avoid writing several tests with the same body of code, consider using the `parameterize` to provide various types of inputs for one function. 
```

**Review by carleyjmartin — APPROVED** [review]
Looks good, just typos
